crates/uv-install-wheel/src/wheel.rs

```diff
@@ -424,16 +424,8 @@ fn install_script(
 
         let mut target = uv_fs::tempfile_in(&layout.scheme.scripts)?;
         let size_and_encoded_hash = copy_and_hash(&mut start.chain(script), &mut target)?;
-        target.persist(&script_absolute).map_err(|err| {
-            io::Error::new(
-                io::ErrorKind::Other,
-                format!(
-                    "Failed to persist temporary file to {}: {}",
-                    path.user_display(),
-                    err.error
-                ),
-            )
-        })?;
+
+        rename_with_retry_sync(&target, &script_absolute)?;
```

Comment:
My only concern is that this is now using rename rather than `persist`. And on Windows, it looks like persist is a bit different, in `tempfile`?

```rust
pub fn persist(old_path: &Path, new_path: &Path, overwrite: bool) -> io::Result<()> {
    unsafe {
        let old_path_w = to_utf16(old_path);
        let new_path_w = to_utf16(new_path);

        // Don't succeed if this fails. We don't want to claim to have successfully persisted a file
        // still marked as temporary because this file won't have the same consistency guarantees.
        if SetFileAttributesW(old_path_w.as_ptr(), FILE_ATTRIBUTE_NORMAL) == 0 {
            return Err(io::Error::last_os_error());
        }

        let mut flags = 0;

        if overwrite {
            flags |= MOVEFILE_REPLACE_EXISTING;
        }

        if MoveFileExW(old_path_w.as_ptr(), new_path_w.as_ptr(), flags) == 0 {
            let e = io::Error::last_os_error();
            // If this fails, the temporary file is now un-hidden and no longer marked temporary
            // (slightly less efficient) but it will still work.
            let _ = SetFileAttributesW(old_path_w.as_ptr(), FILE_ATTRIBUTE_TEMPORARY);
            Err(e)
        } else {
            Ok(())
        }
    }
}
```

Reply:
You're right! I will try to see if I can still use the `persist` function.
Just currently fighting with the borrow-checker, as `persist` takes `self`` and re-returns it in case of `PersistError`,

```rust
error[E0507]: cannot move out of `*from`, as `from` is a captured variable in an `FnMut` closure
   --> crates\uv-fs\src\lib.rs:277:42
    |
262 |     from: &NamedTempFile,
    |     ---- captured outer variable
...
277 |         backoff::retry(backoff, || match from.persist(to) {
    |                                 --       ^^^^ ----------- `*from` moved due to this method call
    |                                 |        |
    |                                 |        move occurs because `*from` has type `NamedTempFile`, which does not implement the `Copy` trait
    |                                 captured by this `FnMut` closure
    |
note: `NamedTempFile::<F>::persist` takes ownership of the receiver `self`, which moves `*from`
   --> C:\Users\Jp\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tempfile-3.14.0\src\file\mod.rs:722:36
    |
722 |     pub fn persist<P: AsRef<Path>>(self, new_path: P) -> Result<F, PersistError<F>> {
```

It's a good rust exercise 😄

Reply:
Updated! Looking forward for your opinion, the signature and the behavior of `persist` was not easy at the beggining to put inside a `backoff::retry` closure,
```rust
pub fn persist<P: AsRef<Path>>(self, new_path: P) -> Result<F, PersistError<F>>
```

> Persist the temporary file at the target path.
>
>If a file exists at the target path, persist will atomically replace it. If this method fails, it will return self in the resulting [PersistError](https://docs.rs/tempfile/latest/tempfile/struct.PersistError.html).

So I found a way to get back the reference returned by the error for every retry.

Looking forward for your opinion on this!